common/flatpak-appdata.c

```diff
@@ -187,6 +192,70 @@ start_element (GMarkupParseContext *context,
             }
         }
     }
+  else if (g_str_equal (element_name, "content_rating"))
+    {
+      /* https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-content_rating */
+      Component *component = NULL;
+
+      g_assert (data->components->len > 0);
+
+      component = g_ptr_array_index (data->components, data->components->len - 1);
+
+      if (component->content_rating == NULL)
+        {
+          const gchar *type = NULL;
+
+          if (g_markup_collect_attributes (element_name,
+                                                 attribute_names,
+                                                 attribute_values,
+                                                 error,
+                                                 G_MARKUP_COLLECT_STRING, "type", &type,
+                                                 G_MARKUP_COLLECT_INVALID))
+            {
+              component->content_rating_type = g_strdup (type);
+              component->content_rating = g_hash_table_new_full (g_str_hash, g_str_equal, NULL, NULL);
```

Comment:
Why _full when both destroynotifies are NULL?

Reply:
It’s a convention I use to make it clear that I have thought about the lifecycle of the key/value and that they are unowned. If I used `g_hash_table_new()`, it wouldn’t be as clear that I’d actually thought about it, rather than forgetting about it.

Reply:
Ok, that is fine to me, just looked a bit weird.